common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```diff
@@ -164,7 +164,7 @@ protected List<Credential> getCredentialsFilteredByInternal(@NonNull final List<
                 && !StringUtil.isNullOrEmpty(authScheme)
                 && credentialType == CredentialType.AccessToken_With_AuthScheme;
         final boolean mustMatchOnKid = !StringUtil.isNullOrEmpty(kid);
-        final boolean mustMatchOnRequestedClaims = !StringUtil.isNullOrEmpty(requestedClaims);
+        final boolean mustMatchOnRequestedClaims = null != requestedClaims;
```

Comment:
So the problem was that ATs with "" no claims where not matched. Is null not also semantically empty? Are there any cases where we need to return all all ATs that differ only in claims? I don;t think MSAL needs this.